File: Python/ByTier/A/2092_A_Kamilka_and_the_Sheep.py
# Problem: CF 2092 A - Kamilka and the Sheep
# https://codeforces.com/contest/2092/problem/A

# ============================================================================
# Problem: A. Kamilka and the Sheep
# Purpose: Find the maximum possible GCD of any two sheep's beauty levels
#          after choosing an optimal non-negative integer d to add to all sheep.
#
# Algorithm:
#   - For each test case, we compute the GCD of all pairs (a_i + d, a_j + d)
#     for different values of d.
#   - A key insight is that increasing all elements by same d does not change
#     pairwise differences. So we can focus on differences between original values,
#     and the problem reduces to choosing a d such that GCD of some pair becomes maximum.
#   - The approach: Try all possible pairs, compute their GCD, and find maximum.
#
# Time Complexity: O(n^2 * log(max(a))) where n is number of sheep
# Space Complexity: O(1) - only using constant extra space
# ============================================================================

for s in [*open(0)][2::2]:  # Read every second line starting from index 2 (skip t and n)
    m, *_, M = sorted(map(int, s.split()))  # Sort the numbers; m=min, M=max
    print(M - m)  # Output difference between max and min values (this is gcd after optimal d)


# https://github.com/VaHiX/codeForces/

File: CSharp/ByTier/A/2092_A_Kamilka_and_the_Sheep.cs
// Problem: CF 2092 A - Kamilka and the Sheep
// https://codeforces.com/contest/2092/problem/A

/*
Problem: A. Kamilka and the Sheep
Algorithm/Techniques: Brute force with GCD calculation
Time Complexity: O(n^2 * log(max_val)) per test case, where n is the number of sheep and max_val is the maximum beauty level.
Space Complexity: O(n) for storing the array of sheep beauty levels.

The problem involves choosing a value d such that when added to all sheep's beauty levels,
the GCD of any two sheep's final beauty levels is maximized. 
Since we can choose d optimally, we explore different values of d and compute the maximum GCD.
However, given the constraints, the approach described here does not require choosing d explicitly;
it just computes the difference between max and min in the original array to determine an optimal scenario,
as the problem reduces to finding two numbers whose GCD is maximized after adding a fixed constant d.

In this code, we read input, compute minimum and maximum of the array (which represents one possible best case),
and output the difference.
*/

using System;
using System.Linq;
class Program {
  static void Main() {
    int t = int.Parse(Console.ReadLine()!); // Read number of test cases
    while (t-- > 0) {
      int n = int.Parse(Console.ReadLine()!); // Read number of sheep
      int[] numbers = 
          Console.ReadLine()!.Split(' ', StringSplitOptions.RemoveEmptyEntries)
              .Select(int.Parse)
              .ToArray(); // Read beauty levels and convert to integer array
      var (minVal, maxVal) = (numbers.Min(), numbers.Max()); // Find min and max values
      Console.WriteLine(maxVal - minVal); // Output the difference (max GCD achieved in optimal d case)
    }
  }
}


// https://github.com/VaHiX/codeForces/